mypy/checker.py

```diff
@@ -5595,6 +5595,58 @@ def has_no_custom_eq_checks(t: Type) -> bool:
                                 and is_overlapping_erased_types(item_type, collection_item_type)
                             ):
                                 if_map[operands[left_index]] = remove_optional(item_type)
+                        else:
+                            operand = operands[right_index]
+                            options: list[LiteralType] = []
+                            is_literal: bool = True
+
+                            iterable_type = get_proper_type(iterable_type)
```

Comment:
I doubt these `isinstance` checks are what we want to be doing (since they force the `get_proper_type` calls everywhere), but I wasn't sure what a better alternative is.